MovingBlocks/Terasology
# Work on OutlineNode and AmbientOcclusionNodes

### Contains
- Refactored OutlineNode to take advantage of the new architecture
- Split AmbientOcclusionPassNode into AmbientOcclusionNode and BlurredAmbientOcclusionNode
- Both new nodes take advantage of the new architecture.
- Created a new SetInputTextureFromFBO state change class: now in use in BlurredAmbientOcclusionNode but couldn't make it work in ApplyDeferredLightingNode [1]
- Minor changes to the RefractiveReflectiveBlocksNode

### How to test
- Run Terasology
- Create a new game or load an existing one
- Verify there have been no changes in the way the 3d world looks


[1] this is an issue with the way the default FBOs are implemented I think. Eventually I plan to implement them differently (not enum-based) and I hope many problems will go away.
